libminifi/src/SchedulingAgent.cpp

```diff
@@ -125,6 +138,17 @@ bool SchedulingAgent::onTrigger(const std::shared_ptr<core::Processor> &processo
   return false;
 }
 
+void SchedulingAgent::watchDogFunc() {
+  std::lock_guard<std::mutex> lock(watchdog_mtx_);
+  auto now = std::chrono::system_clock::now();
+  for (const auto& info : scheduled_processors_) {
+    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - info.start_time_).count();
+    if (elapsed > SCHEDULING_WATCHDOG_ALERT_PERIOD) {
+      logger_->log_warn("%s::onTrigger is running for %zu ms in %s", info.name_, elapsed, info.uuid_);
```

Comment:
`std::chrono::duration::count()` has a return type of `Rep`, which in the case of `std::chrono::milliseconds` is `/*signed integer type of at least 45 bits*/` (https://en.cppreference.com/w/cpp/chrono/duration).
So we can safely assume it's an `int64_t`, but surely not a `size_t`, that the `%zu` would indicate.
I suggest making `elapsed` an `int64_t` and using `%lld` in the format string (`PRId64` would be the completely correct solution, but we don't use these anywhere, that's the scope of a complete refactor).

Reply:
I don't think that any ontrigger call could last more than pow(2,31) millisecs, (that would be at about 23 days), although I agree with the approach :)

Reply:
It's not mainly about representability, but type correctness. With printf it's usually not such a big issue, with scanf it is the source of many security issues, so I like to keep format strings correct.
Also, because the original code cast a signed int to an unsigned one, and because of its usage of non-monotonic clock which could really result in the difference being negative, we would have logged pretty interesting values.